eth/vm/logic/flow.py

```diff
@@ -57,3 +58,36 @@ def gas(computation: BaseComputation) -> None:
     gas_remaining = computation.get_gas_remaining()
 
     computation.stack_push_int(gas_remaining)
+
+
+def beginsub(computation: BaseComputation) -> None:
+    #TODO: raise OOG exception
+    pass
+
+
+def jumpsub(computation: BaseComputation) -> None:
+    sub_loc = computation.stack_pop1_int()
+
+    temp = computation.code.program_counter
+    computation.code.program_counter = sub_loc
+
+    next_opcode = computation.code.peek()
+
+    if next_opcode != BEGINSUB:
```

Comment:
I think you need to first check `computation.code.is_valid_opcode(sub_loc)` to ensure that the position is not part of `PUSHXX` data, then check if `computation.code[sub_loc] == BEGINSUB`, **then** you can set the program counter to the new position safely.

This prevents leaving the code stream in an awkward invalid position in the event of an invalid JUMPSUB.